smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CodegenVisitor.java

```diff
@@ -104,6 +97,21 @@
         // Sort the integrations in specified order.
         integrations.sort(Comparator.comparingInt(TypeScriptIntegration::getOrder));
 
+        // Preprocess model using integrations.
+        for (TypeScriptIntegration integration : integrations) {
+            Model modifiedModel = integration.preprocessModel(context,
```

Comment:
minor: You could do a similar kind of mutation strategy for TypeScriptSettings so that it only needs to be rebuilt each time the model changes.

Reply:
fixed in 16c13c6